src/Engine/Rendering/RenderObject.cpp

```diff
@@ -73,10 +76,12 @@ const std::string& RenderObject::getName() const {
 
 void RenderObject::setVisible( bool visible ) {
     m_visible = visible;
+    invalidateAabb();
 }
 
 void RenderObject::toggleVisible() {
     m_visible = !m_visible;
+    invalidateAabb();
```

Comment:
Not sure about his. The object is not visible but still belong to the scene/engine. So, its aabb might be used to compute the scene aabb no?